engine/index.js

```diff
@@ -303,48 +303,49 @@ const statusFields = ['firefox_status', 'spec_status', 'opera_status',
                       'webkit_status', 'ie_status'];
 // checking for changes in 'status' object
 function checkForNewData(features, dbTestNumber) {
+  let client;
   return redis.getClient(dbTestNumber)
-  .then((client) => new Promise((resolve, reject) => {
-    client.get('status', (err, oldStatus) => {
-      if (err) {
-        console.error('ERROR: ' + err);
-        reject(err, client);
-      }
+  .then((redisClient) => {
+    client = redisClient;
```

Comment:
Nit: Here and in the following functions, if you wrap the rest inside this function you don't need to define a `client` variable.